docs/docs/understanding/workflows/concurrent_execution.md

```diff
@@ -62,6 +85,42 @@ The `step_three` step is fired every time a `StepThreeEvent` is received, but `c
 
 The `result` returned from `collect_events` is an array of the events that were collected, in the order that they were received.
 
+## The `receive` method
+
+In order to make the logic expressed with `collect_event` more user-friendly, we also offer you the possibility of combining the `dispatch` method with the `receive` method:
+
+```python
+class ConcurrentFlow(Workflow):
+    @step
+    async def start(self, ctx: Context, ev: StartEvent) -> StepTwoEvent | None:
+        events = [
+            StepTwoEvent(query="Query 1"),
+            StepTwoEvent(query="Query 2")
+            StepTwoEvent(query="Query 3")
+        ]
+        await ctx.dispatch(events)
+
+    @step(num_workers=4)
+    async def step_two(self, ctx: Context, ev: StepTwoEvent) -> StepThreeEvent:
+        print("Running query ", ev.query)
+        await asyncio.sleep(random.randint(1, 5))
+        return StepThreeEvent(result=ev.query)
+
+    @step
+    async def step_three(
+        self, ctx: Context, ev: StepThreeEvent
+    ) -> StopEvent | None:
+        # wait until we receive 3 events
+        result = await ctx.receive(ev, type(ev))
```

Comment:
nit: I'm not sure `receive` is the most intuitive name here for accumulating multiple events. What about `gather`?

Also, this might be too magical, but the bailing with `None` is kind of confusing. We could perhaps consider throwing a special exception from this call that the workflow infrastructure interprets the same as `None`. Like a `NotReady` exception or something